doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,396 @@
+
+Because of these reasons, 
+it's pretty common to break up large files into smaller chunks.
+For simple data types,
+Pachyderm provides the `--split` flag to `put-file` to automatically do this for you.
+For more complex splitting patterns (e.g. `avro` or other binary formats), 
+you'll need to manually split your data either at ingest or with a Pachyderm pipeline.
+
+#### Split and target-file flags
+For common file types that are often used in data science,
+such as CSV, line-delimited text files, JavaScript Object Notation (json) files, 
+Pachyderm includes the powerful `--split`, `--target-file-bytes` and `--target-file-datums` flags.
+
+`--split`  will divide those files into chunks based on what a "record" is.
+In line-delimited files,
+it's a line.
+In json files,
+it's an object.
+`--split` takes one argument:
+`line`, `json` or [`sql`](../cookbook/splitting.html#pg-dump-sql-support). 
+This argument tell Pachyderm how you want the file split into chunks.
+For example, 
+if you use `--split line`,
+Pachyderm will only divide your file on newline boundaries,
+never in the middle of a line.
+Along with the `--split` flag,
+it's common to use additional "target" flags to get better control over the details of the split.
+Note: We'll call each of the chunks a "split-file" in this document.
+
+
+- `--target-file-bytes` will fill each of the split-files with data up to the number of bytes you specify, splitting on the nearest record boundary.  Let's say you have a line-delimited file of 50 lines, with each line having about 20 bytes.  If you use the flags `--split lines --target-file-bytes 100`, you'll see the input file split into about 10 files or so, each of which will have 5 or so lines.  Each split-file's size will hover above the target value of 100 bytes, not going below 100 bytes until the last split-file, which  may be less than 100 bytes.
+
+- `--target-file-datums` will attempt to fill each split-file with the number of datums you specify.  Going back to that same line-delimited 50-line file above, if you use `--split lines --target-file-datums 2`, you'll see the file split into 50 split-files, each of which will have 2 lines.
+
+- Specifying both flags, `--target-file-datums` and `--target-file-bytes`, will result each split-file containing just enough data to satisfy whichever constraint is hit first.  Pachyderm will split the file and then fill the first target split-file with line-based records until it hits the record limit. If it passes the target byte number with just one record, it will move on to the next split-file.  If it hits the target datum number after adding another line, it will move on to the next split-file. Using the example above, if the flags supplied to put-file are `--split lines --target-file-datums 2 --target-file-bytes 100`, it will have the same result as `--target-file-datums 2`, since that's the most compact constraint, and file sizes will hover around 40 bytes.
```

Comment:
```suggestion
- Specifying both flags, `--target-file-datums` and `--target-file-bytes`, will result in each split-file containing just enough data to satisfy whichever constraint is hit first.  Pachyderm will split the file and then fill the first target split-file with line-based records until it hits the record limit. If it passes the target byte number with just one record, it will move on to the next split-file.  If it hits the target datum number after adding another line, it will move on to the next split-file. Using the example above, if the flags supplied to put-file are `--split lines --target-file-datums 2 --target-file-bytes 100`, it will have the same result as `--target-file-datums 2`, since that's the most compact constraint, and file sizes will hover around 40 bytes.
```

Reply:
This is just changing "will result each" to "will result in each."